lib/compiler-llvm/src/translator/intrinsics.rs

```diff
@@ -39,10 +39,8 @@ pub fn type_to_llvm_ptr<'ctx>(
         Type::F32 => Ok(intrinsics.f32_ptr_ty),
         Type::F64 => Ok(intrinsics.f64_ptr_ty),
         Type::V128 => Ok(intrinsics.i128_ptr_ty),
-        ty => Err(CompileError::Codegen(format!(
-            "type_to_llvm: unimplemented wasmer_types type {:?}",
-            ty
-        ))),
+        Type::FuncRef => Ok(intrinsics.funcref_ty.ptr_type(AddressSpace::Generic)),
+        Type::ExternRef => Ok(intrinsics.funcref_ty.ptr_type(AddressSpace::Generic)),
```

Comment:
Typo? This says `ExternRef` is a pointer to funcref_ty

Reply:
Nope! intrinsics does not have a separate externref_ty. We could all the name and just make it equal to funcref_ty in intrisics.rs?

Reply:
I think it may make sense to be more precise here. I'd be in favor of a `reference_ty` or `anyref_ty` or something to indicate it's an opaque pointer that we don't know / care about the specifics of, when that's the case like in `table.set`.

The main reason is just that we use `funcref_ty` semantically somewhere and it's also a bit confusing to read.

Reply:
We already have an `anyref_ty` and in fact `funcref_ty` is defined as a pointer-to-anyref_ty!

What's an anyref_ty anyways? Is that the right name for what it is?

Reply:
Yeah, I think that name might be confusing. `funcref` should be a pointer to a specific struct. I believe that's called a `VMCallerCheckedAnyfunc` I'd expect an `anyref` to be an opaque pointer. It's probably safe to assume that this opaque pointer at least requires alignment of 1 word, unsure if that really matters, but it should be a safe assumption.